OptimizelySDK.Tests/OdpTests/OdpEventManagerTests.cs

```diff
@@ -0,0 +1,517 @@
+{
+    [TestFixture]
+    public class OdpEventManagerTests
+    {
+        private const string API_KEY = "N0tReAlAp1K3y";
+        private const string API_HOST = "https://odp-events.example.com";
+        private const string MOCK_IDEMPOTENCE_ID = "7d2fc936-8e3b-4e46-aff1-6ccc6fcd1394";
+        private const string FS_USER_ID = "fs_user_id";
+        private const string VUID = "vuid_330e05cad15746d9af8a75b8d10";
+
+        private readonly List<OdpEvent> _testEvents = new List<OdpEvent>
+        {
+            new OdpEvent("t1", "a1", new Dictionary<string, string>
+            {
+                {
+                    FS_USER_ID, "id-key-1"
+                },
+            }, new Dictionary<string, object>
+            {
+                {
+                    "key-1", "value-1"
+                },
+                {
+                    "key-2", null
+                },
+                {
+                    "key-3", 3.3
+                },
+                {
+                    "key-4", true
+                },
+            }),
+            new OdpEvent("t2", "a2", new Dictionary<string, string>
+            {
+                {
+                    FS_USER_ID, "id-key-2"
+                },
+            }, new Dictionary<string, object>
+            {
+                {
+                    "key-2", "value-2"
+                },
+                {
+                    "data_source", "my-source"
+                },
+            }),
+        };
+
+        private readonly List<OdpEvent> _processedEvents = new List<OdpEvent>
+        {
+            new OdpEvent("t1", "a1", new Dictionary<string, string>
+            {
+                {
+                    FS_USER_ID, "id-key-1"
+                },
+            }, new Dictionary<string, object>
+            {
+                {
+                    "idempotence_id", MOCK_IDEMPOTENCE_ID
+                },
+                {
+                    "data_source_type", "sdk"
+                },
+                {
+                    "data_source", Optimizely.SDK_TYPE
+                },
+                {
+                    "data_source_version", Optimizely.SDK_VERSION
+                },
+                {
+                    "key-1", "value-1"
+                },
+                {
+                    "key-2", null
+                },
+                {
+                    "key-3", 3.3
+                },
+                {
+                    "key-4", true
+                },
+            }),
+            new OdpEvent("t2", "a2", new Dictionary<string, string>
+            {
+                {
+                    FS_USER_ID, "id-key-2"
+                },
+            }, new Dictionary<string, object>
+            {
+                {
+                    "idempotence_id", MOCK_IDEMPOTENCE_ID
+                },
+                {
+                    "data_source_type", "sdk"
+                },
+                {
+                    "data_source", Optimizely.SDK_TYPE
```

Comment:
This should be overwritten by custom data.

Reply:
Oh. Ok. The custom data takes priority over the augmented?